### PR TITLE
PR Cleanup and Tests

### PR DESCRIPTION
**Pull request by MondayHopscotch**
Simple tests of each of the 3 formats as well as some fixes exposed while writing the tests.

The .fnt files embedded in the test were generated with the BMFont tool using `Arial Black` as the font, the `@`, `A`, and `T` characters only, and the following font/export settings:

image
image
image

### CODE REVIEW

File: flixel/graphics/frames/bmfont/BMFontInfo.hx [hunk 1]
```diff
@@ -107,17 +107,19 @@ class BMFontInfo
 	static function fromBytes(bytes:BytesInput)
 	{
 		final blockSize = bytes.readInt32();
+		final size = bytes.readInt16();
 		final bitField = bytes.readByte();
+		final charsetByte = bytes.readByte();
```
Comment (MondayHopscotch) [inline flixel/graphics/frames/bmfont/BMFontInfo.hx:112]: We were reading bytes in the wrong order.

File: flixel/graphics/frames/bmfont/BMFontInfo.hx [hunk 2]
```diff
@@ -107,17 +107,19 @@ class BMFontInfo
...
-			size: bytes.readInt16(),
+			size: size,
 			smooth: (bitField & 0x80) != 0,
 			unicode: (bitField & (0x80 >> 1)) != 0,
 			italic: (bitField & (0x80 >> 2)) != 0,
 			bold: (bitField & (0x80 >> 3)) != 0,
 			fixedHeight: (bitField & (0x80 >> 4)) != 0,
-			charset: String.fromCharCode(bytes.readByte()),
-			stretchH: bytes.readInt16(),
+			charset: charsetByte > 0 ? String.fromCharCode(charsetByte) : "",
```
Comment (MondayHopscotch) [inline flixel/graphics/frames/bmfont/BMFontInfo.hx:121]: Prevent getting a charset that is a single character string containing the \0 null character

Comment (UncertainProd) [reply flixel/graphics/frames/bmfont/BMFontInfo.hx:121]: Just out of curiosity, do you know what the charset byte is for? I can't seem to find much info about it online

Comment (MondayHopscotch) [reply flixel/graphics/frames/bmfont/BMFontInfo.hx:121]: It's not super clear exactly what this is for. The text portion of the spec shows:

`charset | The name of the OEM charset used (when not unicode).`

But they don't give a list of values, or where you can get more information on that. You can find pages like [this](https://en.wikipedia.org/wiki/Code_page) that describe "OEM code pages", but their ID's aren't small enough to fit in a single byte... so it's unclear exactly what they mean from the searching I've done.

**Pull request merged**